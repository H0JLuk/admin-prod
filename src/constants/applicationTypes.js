const applicationTypes = [
    {type: 'IOS'},
    {type: 'ANDROID'},
    {type: 'OTHER'}
];

export default applicationTypes;