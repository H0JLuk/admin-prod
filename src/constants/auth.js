export const LOGIN_TYPE = {
    PASSWORD: 'PASSWORD',
    // SBOL_PRO: 'SBOL_PRO',
    SBER_USER_ID: 'SBER_USER_ID',
};