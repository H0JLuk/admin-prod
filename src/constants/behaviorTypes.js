const behaviorTypes = {
    QR: 'QrBehavior',
    OKKO: 'OkkoBehavior',
    SBER_LOGISTIC: 'SberLogisticBehavior',
    DUMMY: 'DummyBehavior',
    EMPTY: 'Empty', //technical behavior to display empty labels while awaiting server response
    WEB: 'WebOnlyBehavior',
};

export default behaviorTypes;
