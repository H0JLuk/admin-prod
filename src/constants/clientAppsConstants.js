export const APP_MECHANICS = {
    PRESENTS: { value: 'PRESENTS', label: 'Подарки' },
    ECOSYSTEM: { value: 'ECOSYSTEM', label: 'Продукты' },
    PRESENTATION: { value: 'PRESENTATION', label: 'Экскурсия' },
    BUNDLE: { value: 'BUNDLE', label: 'WOW' },
};

export const MECHANICS_CHECKBOXES = Object.keys(APP_MECHANICS).map(key => ({
    value: APP_MECHANICS[key].value,
    label: APP_MECHANICS[key].label,
}));

export const APP_LOGIN_TYPES = [
    { label: 'Логин/Пароль', value: 'PASSWORD' },
    { label: 'СБОЛ ПРО', value: 'SBOL_PRO' },
    { label: 'Ссылка', value: 'DIRECT_LINK' },
];

export const MECHANICS_ERROR = `Необходимо выбрать хотя бы одну обязательную механику (${APP_MECHANICS.PRESENTS.label}, ${APP_MECHANICS.ECOSYSTEM.label} или ${APP_MECHANICS.BUNDLE.label})`;
