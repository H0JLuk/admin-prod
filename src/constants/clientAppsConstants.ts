export enum APP_MECHANIC {
    PRESENTS = 'PRESENTS',
    ECOSYSTEM = 'ECOSYSTEM',
    PRESENTATION = 'PRESENTATION',
    BUNDLE = 'BUNDLE',
    EXPRESS = 'EXPRESS',
    MIX = 'MIX',
}

export const APP_MECHANIC_RU: Record<APP_MECHANIC, string> = {
    [APP_MECHANIC.PRESENTS]: 'Подарки',
    [APP_MECHANIC.ECOSYSTEM]: 'Продукты',
    [APP_MECHANIC.PRESENTATION]: 'Экскурсия',
    [APP_MECHANIC.BUNDLE]: 'ВАУ',
    [APP_MECHANIC.EXPRESS]: 'Экспресс',
    [APP_MECHANIC.MIX]: 'Микс',
};

export const APP_MECHANIC_OPTIONS = Object.entries(APP_MECHANIC_RU).map(([value, label]) => ({ label, value }));

export const MECHANICS_ERROR = `Необходимо выбрать хотя бы одну обязательную механику (${APP_MECHANIC_RU.PRESENTS}, ${APP_MECHANIC_RU.ECOSYSTEM}, ${APP_MECHANIC_RU.BUNDLE}, ${APP_MECHANIC_RU.EXPRESS} или ${APP_MECHANIC_RU.MIX})`;

export enum GAME_MECHANIC {
    SCRATCH = 'SCRATCH',
}

export const GAME_MECHANIC_OPTIONS = [
    { label: 'Скретч-слой подарка на экране успеха', value: GAME_MECHANIC.SCRATCH },
];

export enum NOTIFICATION_TYPES {
    PUSH = 'PUSH',
    // SMS: 'SMS',
    // EMAIL: 'EMAIL',
}

export const NOTIFICATION_TYPES_OPTIONS = [
    { label: 'Push', value: NOTIFICATION_TYPES.PUSH },
    // { label: 'SMS', value: NOTIFICATION_TYPES.SMS },
    // { label: 'Email', value: NOTIFICATION_TYPES.EMAIL },
];
