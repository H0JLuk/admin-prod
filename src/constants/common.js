export const showCount = { formatter: ({ count, maxLength }) => `Осталось символов ${maxLength - count}` };

/* '^(https?:\\/\\/)?'+ // protocol
    '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+ // domain name
    '((\\d{1,3}\\.){3}\\d{1,3}))'+ // OR ip (v4) address
    '(\\:\\d+)?(\\/[-a-z\\d%_.~+{}]*)*'+ // port and path
    '(\\?[;&a-z\\d%_.~+=-{}]*)?'+ // query string
    '(\\#[-a-z\\d_]*)?$','i'); // fragment locator */
export const URL_REGEXP = /^(https?:\/\/)?((([a-z\d]([a-z\d-]*[a-z\d])*)\.)+[a-z]{2,}|((\d{1,3}\.){3}\d{1,3}))(:\d+)?(\/[-A-Za-z\d%_.~+#]*)*(\?[;&a-z\d%_.~+=-{}]*)?(#[-a-z\d_]*)?$/;
