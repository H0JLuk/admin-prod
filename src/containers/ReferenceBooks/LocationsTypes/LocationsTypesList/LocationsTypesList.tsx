import React from 'react';
import { Table, TableProps } from 'antd';
import { getFormattedDate } from '@utils/helper';
import { LocationTypeDto } from '@types';

import styles from './LocationsTypesList.module.css';
import EmptyMessage from '@components/EmptyMessage';

const LOCATION_TYPE_NAME = 'Тип локации';
const LOCATION_TYPE_PRIORITY = 'Приоритет типа локации';
const START_DATE = 'Начало действия';
const END_DATE = 'Окончание действия';


type LocationsTypeListProps = Pick<TableProps<LocationTypeDto>, 'onRow' | 'rowSelection' | 'pagination' | 'onChange' | 'loading'> & {
    locationsList: LocationTypeDto[];
};

const LocationsTypesList: React.FC<LocationsTypeListProps> = ({
    locationsList,
    ...restProps
}) => (
    <div className={styles.tableWrapper}>
        <Table<LocationTypeDto>
            {...restProps}
            rowKey="id"
            className={styles.table}
            dataSource={locationsList}
            showHeader={false}
            locale={{ emptyText: <EmptyMessage /> }}
            pagination={false}
        >
            <Table.Column
                width="30%"
                key="name"
                dataIndex="name"
                render={(item) => columnRender(item, LOCATION_TYPE_NAME)}
            />
            <Table.Column
                width="30%"
                key="priority"
                dataIndex="priority"
                render={(item) => columnRender(item, LOCATION_TYPE_PRIORITY)}
            />
            <Table.Column
                width="20%"
                key="startDate"
                dataIndex="startDate"
                render={(item) => columnRender(getFormattedDate(item), START_DATE)}
            />
            <Table.Column
                width="20%"
                key="endDate"
                dataIndex="endDate"
                render={(item) => columnRender(getFormattedDate(item), END_DATE)}
            />
        </Table>
    </div>
);

const columnRender = (item: string, name: string) => (
    <>
        <div className={styles.tableTitle}>{name}</div>
        <div className={styles.tableCell}>{item}</div>
    </>
);

export default LocationsTypesList;
