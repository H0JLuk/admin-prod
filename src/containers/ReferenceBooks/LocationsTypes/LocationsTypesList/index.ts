import LocationsTypesList from './LocationsTypesList';

export default LocationsTypesList;
