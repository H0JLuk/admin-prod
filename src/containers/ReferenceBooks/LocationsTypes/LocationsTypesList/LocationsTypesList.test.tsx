import React from 'react';
import { render } from '@testing-library/react';

import LocationsTypesList from './LocationsTypesList';
import { testLocationsTypesArray } from '../../../../../__tests__/constants';

const props = {
    locationsList: testLocationsTypesArray,
    loading: false,
    onRow: jest.fn(),
    rowSelection: undefined,
};

describe('<LocationsTypesList tests />', () => {
    it('should match snapshot', async () => {
        const { asFragment } = render(<LocationsTypesList {...props} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
