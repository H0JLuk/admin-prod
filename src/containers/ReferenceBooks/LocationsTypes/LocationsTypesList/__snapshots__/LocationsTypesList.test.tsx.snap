// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationsTypesList tests /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="tableWrapper"
  >
    <div
      class="ant-table-wrapper table"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
              >
                <table
                  style="table-layout: auto;"
                >
                  <colgroup>
                    <col
                      style="width: 30%;"
                    />
                    <col
                      style="width: 30%;"
                    />
                    <col
                      style="width: 20%;"
                    />
                    <col
                      style="width: 20%;"
                    />
                  </colgroup>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="0"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        <div
                          class="tableTitle"
                        >
                          Тип локации
                        </div>
                        <div
                          class="tableCell"
                        >
                          Город
                        </div>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <div
                          class="tableTitle"
                        >
                          Приоритет типа локации
                        </div>
                        <div
                          class="tableCell"
                        >
                          6
                        </div>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <div
                          class="tableTitle"
                        >
                          Начало действия
                        </div>
                        <div
                          class="tableCell"
                        >
                          01.01.2020
                        </div>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <div
                          class="tableTitle"
                        >
                          Окончание действия
                        </div>
                        <div
                          class="tableCell"
                        />
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="1"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        <div
                          class="tableTitle"
                        >
                          Тип локации
                        </div>
                        <div
                          class="tableCell"
                        >
                          Город
                        </div>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <div
                          class="tableTitle"
                        >
                          Приоритет типа локации
                        </div>
                        <div
                          class="tableCell"
                        >
                          6
                        </div>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <div
                          class="tableTitle"
                        >
                          Начало действия
                        </div>
                        <div
                          class="tableCell"
                        >
                          01.01.2020
                        </div>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <div
                          class="tableTitle"
                        >
                          Окончание действия
                        </div>
                        <div
                          class="tableCell"
                        />
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="2"
                    >
                      <td
                        class="ant-table-cell"
                      >
                        <div
                          class="tableTitle"
                        >
                          Тип локации
                        </div>
                        <div
                          class="tableCell"
                        >
                          Город
                        </div>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <div
                          class="tableTitle"
                        >
                          Приоритет типа локации
                        </div>
                        <div
                          class="tableCell"
                        >
                          6
                        </div>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <div
                          class="tableTitle"
                        >
                          Начало действия
                        </div>
                        <div
                          class="tableCell"
                        >
                          01.01.2020
                        </div>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <div
                          class="tableTitle"
                        >
                          Окончание действия
                        </div>
                        <div
                          class="tableCell"
                        />
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
